Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.Revolver.cs

```diff
@@ -99,7 +99,24 @@ public bool TryRevolverInsert(EntityUid revolverUid, RevolverAmmoProviderCompone
         if (_whitelistSystem.IsWhitelistFail(component.Whitelist, uid))
             return false;
 
-        // If it's a speedloader try to get ammo from it.
+        // First, check if it's a speedloader and try to get ammo from it.
+        // Otherwise, try to insert the entity directly.
+        if (TryRevolverSpeedLoading(revolverUid, component, uid, user) ||
+            TryRevolverInsertReload(revolverUid, component, uid, user))
+        {
+            UpdateAmmoCount(revolverUid);
+            UpdateRevolverAppearance(revolverUid, component);
+            Dirty(revolverUid, component);
```

Comment:
The Try methods should really be dirtying and handling these changes instead as it shouldn't be on callers to avoid bugs by remembering every method they need to call.